pytensor/tensor/rewriting/math.py

```diff
@@ -423,6 +423,35 @@ def local_sumsqr2dot(fgraph, node):
                     return [new_out]
 
 
+@register_canonicalize
+@register_specialize
+@node_rewriter([Elemwise])
```

Comment:
This will make it so the rewrite is only called on these nodes, and you can skip the first `isinstance` check

```suggestion
@node_rewriter([mul, true_div])
```